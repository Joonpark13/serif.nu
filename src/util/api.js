/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
