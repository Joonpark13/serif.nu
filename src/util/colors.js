export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
