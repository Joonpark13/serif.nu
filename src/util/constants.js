/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
