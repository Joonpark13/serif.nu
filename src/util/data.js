import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
