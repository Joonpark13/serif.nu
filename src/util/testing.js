/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
