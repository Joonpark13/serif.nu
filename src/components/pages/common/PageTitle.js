import React from 'react';
import { northwesternPurple10 } from 'util/colors';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import Typography from '@material-ui/core/Typography';

export const styles = {
  page: {
    margin: 20,
    padding: 5,
    borderBottom: `2px solid ${northwesternPurple10}`,
  },
};


function PageTitle({ classes, title }) {
  return (
    <div className={classes.page}>
      <Typography variant="h3" color="primary">
        {title}
      </Typography>
    </div>
  );
}

PageTitle.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  title: PropTypes.string.isRequired,
};

export { PageTitle as UnstyledPageTitle };
export default withStyles(styles)(PageTitle);
