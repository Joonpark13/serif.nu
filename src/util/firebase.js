import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
