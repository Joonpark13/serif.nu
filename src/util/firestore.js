import firebase from 'firebase/app';
import 'firebase/firestore';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export default firebase.firestore();
