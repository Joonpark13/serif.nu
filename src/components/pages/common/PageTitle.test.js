import { wrapperCreator } from 'util/testing';
import { UnstyledPageTitle, styles } from './PageTitle';

describe('PageTitle', () => {
  const defaultProps = {
    title: 'Title',
  };
  const getComponent = wrapperCreator(UnstyledPageTitle, defaultProps, styles);

  it('renders correctly', () => {
    const wrapper = getComponent();

    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
