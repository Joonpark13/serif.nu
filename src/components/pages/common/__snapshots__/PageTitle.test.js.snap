// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle renders correctly 1`] = `
<div
  className=""
>
  <WithStyles(Typography)
    color="primary"
    variant="h3"
  >
    Title
  </WithStyles(Typography)>
</div>
`;
